docs/how-to/code/internal-user/rockcraft.yaml

```diff
@@ -1,32 +1,36 @@
 name: internal-user
-base: bare
-build-base: ubuntu@24.04
+base: ubuntu@22.04
 version: 'latest'
 summary: Add an internal user inside a rock
 description: Add an internal user inside a rock, and run a service with it.
 platforms:
-    amd64:
+  amd64:
 
 # [docs:rock-services]
 services:
-    whoami:
-        override: replace
-        startup: enabled
-        command: bash -c "whoami; sleep 5"
-        user: ubuntu
-        on-success: ignore 
+  web-service:
+    override: replace
+    command: node server.js
+    startup: enabled
+    on-success: shutdown
+    on-failure: shutdown
+    working-dir: /lib/node_modules/node_web_app
+    user: myuser
```

Comment:
Whilst I understand that this is a more proper service to exemplify, is complicates the guide a lot, making it more about Node than about the adding of a user. We need a simpler test service. Or even no service at all, and then just test the rock with `pebble exec --user=muser whoami`. If you have a simple proposal of a basic service that could print this username, without adding too much complexity to this guide, pls share

Reply:
Let me recap what I tried so far before picking the node server example above:

1. I tried in `base: bare` a service just "python3.12 -m http.server --cgi" with a simple python script in `cgi-bin` folder that returns the username in a HTTP response. Result: it worked locally, but in spread runner, it seems that the python http server wasn't able to listen on the specified port (tried different ports >= 8000). When curl tried to access the url (something like http://127.0.0.1:8000/cgi-bin/getuser.py) it got "connection reset by peer" error. The docker container was run with `-p 8000:8000`).

2. Next, I tried a python script that use the `http.server` module to launch an HTTP server, but instead of sending the username in an HTTP response which couldn't be tested in a spread runner apparently, I just printed out the username in stdout. Result: nothing was printed out in stdout. Apparently, the bare base with just base-files and base-passwd is missing something that makes python unable to print to stdout in certain contexts (the context of a http service).

3. So as not to overcomplicate the example of the service, I switched to the node example we have in our tutorials. However, after trying it, it doesn't work with bare base neither, so I changed the base accordingly.

Reply:
(discussed offline - let's try2)

Reply:
changed the test to a python http server.